tests/90jira/SYN-606.pl

```diff
@@ -27,20 +26,28 @@
          })->then( sub {
             matrix_initialsync_room( $nonjoined_user, $room_id1 )
          })->then( sub {
+            my ( $body ) = @_;
+
+            my $from_token = $body->{messages}{end};
+
             Future->needs_all(
                matrix_send_room_text_message( $user, $room_id1, body => "moose" ),
-               await_event_not_history_visibility_or_presence_for( $nonjoined_user, $room_id1, [] ),
+               await_event_not_history_visibility_or_presence_for( $nonjoined_user, $room_id1, [],
+                  from => $from_token,
```

Comment:
there's no explanation anywhere (PR, commit comment, a comment in code) as to what is going on here or why it is a good thing. Am I supposed to read your mind?

generally passing an explicit `from` token seems a bit odd here. I thought the whole point of the `await_event` functions was that they took this stuff away from you (and if you wanted to manage it yourself, you ought to be using lower-level methods).

If this is the right thing to e doing for some reason, please could you add a comment explaining why.

Reply:
Sorry, I've updated with a comment.